src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb

```diff
@@ -571,6 +569,18 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                 tList.Add(MakeWhiteSpaceTrivia(GetText(Here - 1)))
             End If
 
+            If foundComment Then
+                Dim comment As SyntaxTrivia = ScanComment()
+                If Not CheckFeatureAvailability(Feature.CommentsAfterLineContinuation) Then
+                    comment = comment.WithDiagnostics({ErrorFactory.ErrorInfo(ERRID.ERR_CommentsAfterLineContinuationNotAvailable1,
+                        New VisualBasicRequiredLanguageVersion(Feature.CommentsAfterLineContinuation.GetLanguageVersion()))})
+                End If
+                tList.Add(comment)
+                ' Need to call CanGet here to prevent Peek reading past EndOfBuffer. This can happen when file ends with comment but no New Line.
+                ch = If(CanGet(), Peek(), ChrW(0))
```

Comment:
>ch = If(CanGet(), Peek(), ChrW(0)) [](start = 16, length = 34)

Why do the assignment and execute the next statement when ```CanGet()``` returns false?

Reply:
Why assign to ```ch``` at all at this point?

---
In reply to: [217788133](https://github.com/dotnet/roslyn/pull/28327#discussion_r217788133) [](ancestors = 217788133)

Reply:
I was having same conversion with Julien both work, I thought this is shorter, happy to change.

Reply:
You still need to set atNewLine either to IsNewLine of False.

Reply:
This also works but is it better?
```
               If CanGet() Then
                    ch = Peek()
                    atNewLine = IsNewLine(ch)
                Else
                    atNewLine = False
                End If
 ```

Reply:
> I was having same conversion with Julien both work, I thought this is shorter, happy to change.

I, personally, value shortness of the code the least. Clarity, what it is doing and why is way more important.

---
In reply to: [217790089](https://github.com/dotnet/roslyn/pull/28327#discussion_r217790089) [](ancestors = 217790089)

Reply:
> This also works but is it better?

This is definitely better.  Why the ```Else``` portion is needed though? Isn't ```atNewLine``` already false?

---
In reply to: [217791770](https://github.com/dotnet/roslyn/pull/28327#discussion_r217791770) [](ancestors = 217791770)

Reply:
Changed